v21.1/stream-data-out-of-cockroachdb-using-changefeeds.md

```diff
@@ -96,6 +96,10 @@ The `kv.closed_timestamp.target_duration` [cluster setting](cluster-settings.htm
     Resolved timestamp notifications on every Kafka partition can be used to provide strong ordering and global consistency guarantees by buffering records in between timestamp closures. Use the "resolved" timestamp to see every row that changed at a certain time.
 
     The complexity with timestamps is necessary because CockroachDB supports transactions that can affect any part of the cluster, and it is not possible to horizontally divide the transaction log into independent changefeeds. For more information about this, [read our blog post on CDC](https://www.cockroachlabs.com/blog/change-data-capture/).
+    
+## Delete messages
+
+Deleting a row will result in a changefeed message giving the primary key of the deleted row and a null value. If an attempt is made to delete a row that never existed, changefeed behavior is undefined to allow for optimizations at the storage layer: you may or may not get a delete message. Similarly, creating and deleting a row within the same transaction will never result in an insert message, but may result in a delete message.
```

Comment:
Maybe an example of what a delete message looks like, something like {"id": 1, "name": null}

I would also just add a sentence that makes it clear what the impact of the undefined behavior. Something like - "In some situations you may get a delete message for an ID that never existed" then the rest - "If an attempt is made to delete a row that never existed, changefeed behavior is undefined to allow for optimizations at the storage layer: you may or may not get a delete message."

For the last part- I would add an explanation of what guarentees we have around transaction, something like -  "We only propagate the last change to a row in a transaction through a changefeed", then "Creating and deleting a row within the same transaction will never result in an insert message, but may result in a delete message."

Reply:
Added, thanks!